lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -671,26 +689,32 @@ def _build_student_data(cls, user_id, course_key, usage_key_str):
                 if hasattr(block, 'generate_report_data'):
                     try:
                         user_state_iterator = user_state_client.iter_all_for_block(block_key)
-                        generated_report_data = {
-                            username: state
-                            for username, state in
-                            block.generate_report_data(user_state_iterator, max_count)
-                        }
+                        generated_report_data = dict()
+                        for username, state in block.generate_report_data(user_state_iterator, max_count):
+                            generated_report_data.setdefault(username, []).append(state)
```

Comment:
Nit: This is fine as is, but I find `collections.defaultdict` slightly easier to read:

```python
generated_report_data = defaultdict(list)
for username, state in block.generate_report_data(user_state_iterator, max_count):
    generated_report_data[username].append(state)
````

This is a stylistic choice and completely optional. I just wanted to mention it in case you weren't familiar with `defaultdict`.